modules/renter/streamer.go

```diff
@@ -0,0 +1,141 @@
+package renter
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math"
+	"time"
+
+	"github.com/NebulousLabs/errors"
+)
+
+type (
+	// streamer is a io.ReadSeeker that can be used to stream downloads from
+	// the sia network.
+	streamer struct {
+		file   *file
+		offset int64
+		r      *Renter
+	}
+)
+
+// min is a helper function to find the minimum of multiple values.
+func min(values ...uint64) uint64 {
+	min := uint64(math.MaxUint64)
+	for _, v := range values {
+		if v < min {
+			min = v
+		}
+	}
+	return min
+}
+
+// Streamer create a io.ReadSeeker that can be used to stream downloads from
+// the sia network.
+func (r *Renter) Streamer(siaPath string) (string, io.ReadSeeker, error) {
+	// Lookup the file associated with the nickname.
+	lockID := r.mu.RLock()
+	file, exists := r.files[siaPath]
+	r.mu.RUnlock(lockID)
+	if !exists {
+		return "", nil, fmt.Errorf("no file with that path: %s", siaPath)
+	}
+	// Create the streamer
+	s := &streamer{
+		file: file,
+		r:    r,
+	}
+	return file.name, s, nil
+}
+
+// Read implements the standard Read interface. It will download the requested
+// data from the sia network and block until the download is complete.  To
+// prevent http.ServeContent from requesting too much data at once, Read can
+// only request a single chunk at once.
+func (s *streamer) Read(p []byte) (n int, err error) {
+	// Get the file's size
+	s.file.mu.RLock()
+	fileSize := int64(s.file.size)
+	s.file.mu.RUnlock()
+
+	// Make sure we haven't reached the EOF yet.
+	if s.offset >= fileSize {
+		return 0, io.EOF
+	}
+
+	// Calculate how much we can download. We never download more than a single chunk.
+	chunkSize := s.file.staticChunkSize()
+	remainingData := uint64(fileSize - s.offset)
+	requestedData := uint64(len(p))
+	remainingChunk := chunkSize - uint64(s.offset)%chunkSize
+	length := min(chunkSize, remainingData, requestedData, remainingChunk)
+
+	// Download data
+	buffer := bytes.NewBuffer([]byte{})
+	d, err := s.r.newDownload(downloadParams{
+		destination:       newDownloadDestinationWriteCloserFromWriter(buffer),
+		destinationType:   destinationTypeSeekStream,
+		destinationString: "httpresponse",
+		file:              s.file,
+
+		latencyTarget: 25e3 * time.Millisecond, // TODO high default until full latency suport is added.
```

Comment:
Since the user is going to be blocking on this call until the download completes, a low latency target is going to be optimal. I'd place the default here at 50ms.